controllers/deployment/infrastructure_status.go

```diff
@@ -0,0 +1,163 @@
+package deployment
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/netip"
+	"slices"
+	"strings"
+
+	v2 "github.com/metal-stack/firewall-controller-manager/api/v2"
+	"github.com/metal-stack/firewall-controller-manager/controllers"
+	"github.com/metal-stack/metal-lib/pkg/pointer"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (c *controller) updateInfrastructureStatus(r *controllers.Ctx[*v2.FirewallDeployment], infrastructureName string, ownedFirewalls []*v2.Firewall) error {
+	infraObj := &unstructured.Unstructured{}
+
+	infraObj.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "extensions.gardener.cloud",
+		Kind:    "Infrastructure",
+		Version: "v1alpha1",
+	})
+
+	err := c.c.GetSeedClient().Get(r.Ctx, client.ObjectKey{
+		Namespace: c.c.GetSeedNamespace(),
+		Name:      infrastructureName,
+	}, infraObj)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	type infrastructure struct {
```

Comment:
As far as I know for the compiler it does not make a difference because a struct is always constant, no matter where it is defined (there is no dynamic struct generation). Only difference should be the scope and I do not want other functions to re-use this. Actually the Gardener-related stuff should be as isolated as possible.